src/datatype/svecf32.rs

```diff
@@ -0,0 +1,321 @@
+use crate::prelude::*;
+use pgrx::pg_sys::Datum;
+use pgrx::pg_sys::Oid;
+use pgrx::pgrx_sql_entity_graph::metadata::ArgumentError;
+use pgrx::pgrx_sql_entity_graph::metadata::Returns;
+use pgrx::pgrx_sql_entity_graph::metadata::ReturnsError;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlMapping;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlTranslatable;
+use pgrx::FromDatum;
+use pgrx::IntoDatum;
+use service::prelude::*;
+use std::alloc::Layout;
+use std::cmp::Ordering;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ops::Index;
+use std::ops::IndexMut;
+use std::ptr::NonNull;
+
+#[repr(C, align(8))]
+pub struct SVecf32 {
```

Comment:
We may need to track dimensions of a vector.

Reply:
Current implementation is to ignore dimensions in `SVef32` to support dynamic dimension like `vector`. When casting to `vector`, users must set the dimension explicitly. And when passing a sparse vector to the backend, it will check the last element.

https://github.com/silver-ymz/pgvecto.rs/blob/feat/sparse/crates/service/src/prelude/storage/sparse.rs#L54-L56

Reply:
I think it's not a good way. Let's record dimensions of a vector in this structure. We can still implement "dynamic dimension" as long as we do not check typmod. It's consistent with semantics of `vector`.

Reply:
added in new commit